spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ArgumentsNamespacingStrategy.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.launcher;
+
+import java.util.Map;
+
+import org.springframework.core.env.PropertySources;
+
+
+/**
+ * Encapsulates the operation of isolating arguments for different modules. Provides two operations that are the
+ * inverse of each other. That is, the following pseudo invariants must hold:
+ * <pre>
+ *     props == unqualify("foo", qualify("foo", props))
+ *     unqualify("foo", qualify("bar", props)) == empty
+ * </pre>
+ *
+ * @author Eric Bottard
+ */
+public interface ArgumentsNamespacingStrategy {
```

Comment:
Yes, I realized that once I got it in. We _may_ want something about escaping the colons (but from separate comment, they may go altogether anyway), or turning things uppercase.